### ISSUE TITLE
RKE2 Provisioning: Day 2 Ops - On Demand Etcd Snapshot and Restore (Local Snapshots)

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#32391]
Support for Local snapshot and Restore Use case

```
apiVersion: provisioning.cattle.io/v1
kind: Cluster
metadata:
  name: #string
  annotations:
    {}
#    key: string
  labels:
    {}
#    key: string
  namespace: fleet-default
spec:
  cloudCredentialSecretName: do
  kubernetesVersion: v1.21.0-alpha2+rke2r1
  rkeConfig:

#    etcdSnapshotCreate:
#      name: string
#      nodeName: string
#      s3:
#        bucket: string
#        cloudCredentialName: string
#        endpoint: string
#        endpointCA: string
#        folder: string
#        region: string
#        skipSSLVerify: boolean
```

**Comment by anupama2501** [comment]
On master-744c23307d61bd5e0951a1a6c76f32335f8011fa-head
1. From cluster explorer, created a node driver cluster and from edit as yaml and provided the following config
etcdSnapshotCreate:
      name: test-anu
Cluster creation has failed with the error:
`failed to find node to perform etcd snapshot`
2. From cluster explorer, created a node driver cluster and once the node is active from edit as yaml and provided the following config
etcdSnapshotCreate:
    etcdSnapshotCreate:
      name: anu-snapshot      

<img width="519" alt="Screen Shot 2021-05-25 at 1 58 24 PM" src="https://user-images.githubusercontent.com/38144301/119567540-4b6f6100-bd61-11eb-84a8-2e67439eea3d.png">

**Comment by anupama2501** [comment]
Re-opening the ticket for testcase1: 

> From cluster explorer, created a node driver cluster and from edit as yaml and provided the following config
etcdSnapshotCreate:
name: test-anu
Cluster creation has failed with the error:
failed to find node to perform etcd snapshot

**Comment by anupama2501** [comment]
Keeping this issue in to-test for further validations and creating new tickets for issues seen with etcd snapshot rke2

**Comment by deniseschannon** [comment]
@anupama2501 The etcd object is changing as of yesterday, so please hold off on testing.

**Comment by deniseschannon** [comment]
@anupama2501 Never mind, this yaml is still valid and can be tested using master.

**Comment by anupama2501** [comment]
On master-head 7d5a854
1. Create a multinode cluster 1 node cp + worker and 3 etcd nodes
2. Once the cluster is active, from cluster management --> cluster --> Snapshots --> Snapshot Now --> node on which the snapshot is saved goes into provisioning state and stuck in that state.

Result:
The node shouldnot go into provisioning state when a snapshot is taken. Snapshots should've been saved on the node. Snapshots are not created and not saved on the nodes.

Error status in the yaml:
```
    message: waiting for etcd snapshot
    reason: Waiting
    status: Unknown
    type: Provisioned
```
Errors in the logs:
```
2021/06/02 22:32:16 [DEBUG] [etcd-backup] checking backups for cluster [c-m-8218c498]
2021/06/02 22:32:16 [DEBUG] [etcd-backup] [c-m-8218c498] is not an rke cluster, skipping..
2021/06/02 22:32:16 [DEBUG] DesiredSet - No change(2) management.cattle.io/v3, Kind=Cluster /c-m-8218c498 for cluster-create fleet-default/anu-multietc2
2021/06/02 22:32:16 [DEBUG] DesiredSet - No change(2) management.cattle.io/v3, Kind=ClusterRoleTemplateBinding c-m-8218c498/c-m-8218c498-owner for cluster-create fleet-default/anu-multietc2
2021/06/02 22:32:17 [DEBUG] DesiredSet - No change(2) management.cattle.io/v3, Kind=Cluster /c-m-8218c498 for cluster-create fleet-default/anu-multietc2
2021/06/02 22:32:17 [DEBUG] DesiredSet - No change(2) management.cattle.io/v3, Kind=ClusterRoleTemplateBinding c-m-8218c498/c-m-8218c498-owner for cluster-create fleet-default/anu-multietc2

```


cluster yaml:
[anu-multietc2 .txt](https://github.com/rancher/rancher/files/6587754/anu-multietc2.txt)

**Comment by deniseschannon** [comment]
@kinarashah Please add in tests for creating and restoring snapshots

**Comment by deniseschannon** [comment]
In theory, all that's needed to take a snapshot is a name.

If you don't select a nodename, it should go through the list of nodes to find the etcd node and pick one of those.

**Comment by thedadams** [comment]
In order to create a snapshot, the node plan needs to be updated. The status of the node gets updated during this process. I think it makes sense for this status to be `Waiting` instead of `Provisioned`, but the status will get updated.

**Comment by davidnuzik** [comment]
**Reproduction:**
Not required.

**Validation Successful:**
Rancher version: master-head `30574c2a9`
Rancher cluster type: single-node docker container
Docker version: 19.03

Downstream cluster type: RKE2 custom cluster (Digital Ocean provider)
Downstream K8s version: v1.21.3-rc1+rke2r2

_**Use Case 0 - Local snapshots (simple 1 node cluster)**_
Validation steps:
_This will be a quick test just to ensure nothing is seriously broken before I try the more complicated use cases._
1. Provision a simple RKE2 cluster with rancher. Make this just a 1-node cluster with all roles. _Use the default settings_ (which enables automatic snapshots with `0 */5 * * *` cron schedule [every 5 hours] and keeping the last 5 snapshots).
2. When the cluster is ready, go to the cluster details and click the "Snapshot Now" button under the "Snapshots" tab:
![image](https://user-images.githubusercontent.com/45179589/128072742-cd2f9cef-0dd5-4641-a978-15854ee2e5b9.png)
The button reads as "Snapshoting.." and then turns green and says "Snapshot creating" and then after a short time my snapshot is ready. I see the snapshot listed and as Active as pictured:
![image](https://user-images.githubusercontent.com/45179589/128072860-ca991580-49f9-44eb-b3e9-467b295002ec.png)
3. Deploy a workload, such as nginx:latest as a Deployment just to make a change to the cluster. It successfully deploys.
4. Restore the snapshot I took. The UI asks me to confirm with an in-window pop-up. The date and time are correct (my local date/time I took the snapshot)
![image](https://user-images.githubusercontent.com/45179589/128073285-ec26fae0-b097-469f-b695-789bd928c024.png)
I Click on Restore and see the button change to "Restoring.." briefly then the pop-up is dismissed. The cluster goes into a Provisioning state (since the cluster is restoring).
I check the status on the cluster manager page and after a short time I see an error that the cluster health check failed (if this is temporary I would expect the UI to suppress this for a time so as to not alarm the user):
![image](https://user-images.githubusercontent.com/45179589/128073670-239815a7-1a03-40cb-8564-caa9273272ae.png)
This is what I see in the rancher server logs during this restore process:
_Note, I removed some irrelevant entries from the logs inbetween these lines like websocket disconnects, etc_
```
2021/08/03 19:18:15 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: refreshing etcd restore state                                                                                                                                                           
2021/08/03 19:18:15 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: refreshing etcd restore state                                                                                                                                                           
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: shutting down control plane                                                                                                                                                             
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]                                                                                             
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]                                                                                             
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]                                                                                             
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]                                                                                             
2021/08/03 19:18:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]
2021/08/03 19:18:31 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting to shutdown down control plane node [dave-testrke2snapshots-c1-simple-simple-6gg5f]
2021/08/03 19:18:31 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: refreshing etcd restore state
2021/08/03 19:18:31 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: starting etcd restore
2021/08/03 19:18:31 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting for etcd restore
2021/08/03 19:18:31 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting for etcd restore
2021/08/03 19:18:32 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting for etcd restore
2021/08/03 19:18:32 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting for etcd restore
2021/08/03 19:19:24 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:20:11 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:20:15 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: waiting for etcd restore
2021/08/03 19:20:15 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: refreshing etcd restore state
2021/08/03 19:20:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm
2021/08/03 19:20:16 [INFO] rate limited mcc-bundle(map[objectset.rio.cattle.io/hash:7f8667c9c99c99f1b6f3909eebeb7ffd9f11ac0d]) 1.010990123s
2021/08/03 19:20:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm
2021/08/03 19:20:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm
2021/08/03 19:20:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting for plan to be applied
2021/08/03 19:20:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting for plan to be applied
2021/08/03 19:21:02 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:21:47 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:22:34 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:23:21 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:24:08 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:24:53 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:25:37 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting for plan to be applied
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting for plan to be applied
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: non-ready bootstrap machine(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting for plan to be applied
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico, kube-controller-manager, kube-scheduler
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico, kube-controller-manager, kube-scheduler
2021/08/03 19:26:11 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico, kube-controller-manager, kube-scheduler
2021/08/03 19:26:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico, kube-controller-manager, kube-scheduler
2021/08/03 19:26:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico, kube-controller-manager, kube-scheduler
2021/08/03 19:26:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico
2021/08/03 19:26:16 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: provisioning bootstrap node(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico
2021/08/03 19:26:21 [ERROR] error syncing 'c-m-b5253925': handler cluster-deploy: Get "https://10.43.0.1:443/apis/apps/v1/namespaces/cattle-system/daemonsets/cattle-node-agent": cluster agent disconnected, requeuing
2021/08/03 19:26:56 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: non-ready bootstrap machine(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico                                                                      
2021/08/03 19:26:56 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: non-ready bootstrap machine(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico                                                                      
2021/08/03 19:26:56 [INFO] rkecluster fleet-default/dave-testrke2snapshots-c1-simple: non-ready bootstrap machine(s) dave-testrke2snapshots-c1-simple-simple-785f55c9c6-r79tm: waiting on probes: calico                                                                      
2021/08/03 19:26:58 [INFO] Handling backend connection request [c-m-b5253925]
```
All in all it took approximately 7 minutes for the restore to complete for this 1-node rke2 cluster. It did complete successfully, but during the restoration process I saw that error in the UI making me think something is wrong. We should consider changing this behavior later, but I believe this is no different than how things have worked historically such as with the ember UI. If I were an end-user and didn't know this was normal I would be panicking.
5. Explore the cluster and look at workload deployments. The nginx deployment is now gone, as expected, since the etcd snapshot restoration was successful. (Note, due to a [bug](https://github.com/rancher/dashboard/issues/3623) you may need to manually refresh the Deployments page).

Overall, the process was successful with use case 0. This simple and quick check ensures that a 1-node cluster with local snapshots restores successfully.

_**Use Case 1 - Local snapshots FAILED DUE TO KNOWN BUG https://github.com/rancher/rancher/issues/32362**_
Validation steps:

Perform the same steps as in Use Case 0, except, create a more "complicated" cluster like:
- 3 etcd nodes in one pool
- 2 control plane nodes in another pool
- 3 worker nodes in another pool
_**I am blocked by bug https://github.com/rancher/rancher/issues/32362 - cannot provision such clusters**_

_**Use Case 2 - s3 bucket snapshots - FAILED**_
Validation steps:
Perform the same steps as in use Case 1, except during cluster creation, specify s3 etcd backups in the UI.
_**Note due to bug https://github.com/rancher/rancher/issues/32362 I tested with 1 node with all roles and found this issue:**_

During testing I noticed anything passed into s3 config for the cluster does not go thru. Here is the POST request body for creating the cluster:
```
{
	"metadata": {
		"annotations": {
			"field.cattle.io/description": "Does the s3 info even get passed upon rk2 cluster creation? I didn't see it in cluster yaml!"
		},
		"name": "dave-test-s3-again",
		"namespace": "fleet-default"
	},
	"spec": {
		"cloudCredentialSecretName": "dave-rke2-do-creds",
		"defaultPodSecurityPolicyTemplateName": "",
		"kubernetesVersion": "v1.21.3-rc1+rke2r2",
		"rkeConfig": {
			"chartValues": {
				"rke2-calico": {
					"calicoctl": {
						"image": "rancher/mirrored-calico-ctl",
						"tag": "v3.19.1"
					},
					"certs": {
						"node": {
							"cert": null,
							"commonName": null,
							"key": null
						},
						"typha": {
							"caBundle": null,
							"cert": null,
							"commonName": null,
							"key": null
						}
					},
					"global": {
						"systemDefaultRegistry": ""
					},
					"imagePullSecrets": {},
					"installation": {
						"calicoNetwork": {
							"bgp": "Disabled",
							"ipPools": [
								{
									"blockSize": 24,
									"cidr": "10.42.0.0/16",
									"encapsulation": "VXLAN",
									"natOutgoing": "Enabled"
								}
							]
						},
						"controlPlaneTolerations": [
							{
								"effect": "NoSchedule",
								"key": "node-role.kubernetes.io/control-plane",
								"operator": "Exists"
							},
							{
								"effect": "NoExecute",
								"key": "node-role.kubernetes.io/etcd",
								"operator": "Exists"
							}
						],
						"enabled": true,
						"imagePath": "rancher",
						"imagePrefix": "mirrored-calico-",
						"kubernetesProvider": ""
					},
					"ipamConfig": {
						"autoAllocateBlocks": true,
						"strictAffinity": true
					},
					"tigeraOperator": {
						"image": "rancher/mirrored-calico-operator",
						"registry": "docker.io",
						"version": "v1.17.4"
					}
				}
			},
			"etcd": {
				"disableSnapshots": false,
				"s3": null,
				"snapshotRetention": 5,
				"snapshotScheduleCron": "0 */5 * * *"
			},
			"localClusterAuthEndpoint": {
				"caCerts": "",
				"enabled": false,
				"fqdn": ""
			},
			"machineGlobalConfig": {
				"cni": "calico",
				"profile": null
			},
			"machinePools": [
				{
					"controlPlaneRole": true,
					"etcdRole": true,
					"hostnamePrefix": "dave-test-s3-again-pool1-",
					"labels": {},
					"machineConfigRef": {
						"kind": "DigitaloceanConfig",
						"name": "nc-dave-test-s3-again-pool1-h8cc5"
					},
					"name": "pool1",
					"quantity": 1,
					"workerRole": true
				}
			],
			"machineSelectorConfig": [
				{
					"config": {
						"cloud-provider-name": "none"
					}
				}
			],
			"upgradeStrategy": {
				"controlPlaneConcurrency": "10%",
				"controlPlaneDrainOptions": {},
				"workerConcurrency": "10%",
				"workerDrainOptions": {}
			}
		}
	},
	"type": "provisioning.cattle.io.cluster"
}
```

I had specified the region, bucket, aws credentials, etc and none of this is showing. S3 backups don't seem to work as the s3 params are not passed by the UI upon submission (POST) to `provisioning.cattle.io.clusters` "s3" should not be null.

**Comment by davidnuzik** [comment]
BLOCKED by https://github.com/rancher/dashboard/issues/3624 and https://github.com/rancher/rancher/issues/32362

**Comment by davidnuzik** [comment]
Blocked by bug https://github.com/rancher/rancher/issues/33950 (Use Case 2 issue in my prior comment).

**Comment by davidnuzik** [comment]
**Reproduction:**
Not required.

**Validation Failed:**
Rancher version: v2.6-head `246fbd5fa` 8/5/21
Rancher cluster type: single-node docker container
Docker version: 19.03

Downstream cluster type: RKE2 rancher-provisioned cluster (Digital Ocean provider)
Downstream K8s version: v1.21.3-rc3+rke2r2

Validation steps:
Use Case 2 - s3 bucket snapshots FAILED
_**Use Case 0 and 1 already passed with prior testing (local etcd snapshots)**_
1. Create a cluster. Submit s3 information.
2. Take a snapshot. The UI is stuck showing "waiting for plan to be applied" and I also see the same in the rancher server side - the rancher server doesn't update with new messages.

```
2021/08/05 22:35:57 [INFO] rkecluster fleet-default/dave-rke2-s3test: refreshing etcd create state
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: starting etcd snapshot
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:35:58 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: waiting for etcd snapshot
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: refreshing etcd create state
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: provisioning bootstrap node(s) dave-rke2-s3test-e-59bc958d8-bvkrg
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: provisioning bootstrap node(s) dave-rke2-s3test-e-59bc958d8-bvkrg
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: provisioning bootstrap node(s) dave-rke2-s3test-e-59bc958d8-bvkrg: waiting for plan to be applied
2021/08/05 22:36:04 [INFO] rkecluster fleet-default/dave-rke2-s3test: provisioning bootstrap node(s) dave-rke2-s3test-e-59bc958d8-bvkrg: waiting for plan to be applied
2021/08/05 22:36:22 [INFO] rkecluster fleet-default/dave-rke2-s3test: provisioning bootstrap node(s) dave-rke2-s3test-e-59bc958d8-bvkrg: waiting for plan to be applied
```

3. Inspect the s3 bucket - everything is there it looks like the snapshot was successful.
4. Inspect the snapshots tab - none of the snapshots show. The UI likely thinks the snapshots failed probably because the backend didn't provide a message that the snapshot completed.

I've created a separate issue to track this problem here: https://github.com/rancher/dashboard/issues/3690
I checked and downstream rke1 doesn't seem to have this issue.

I will move this issue back To Test again while we await a fix for this new bug to unblock testing s3 snapshots here.

**Comment by davidnuzik** [comment]
Update: basically blocked by https://github.com/rancher/rke2/issues/1551 which is actively being worked on.

**Comment by davidnuzik** [comment]
Update: https://github.com/rancher/rke2/issues/1551 required a lot of troubleshooting/debugging but have an understanding as to the nature of the issue now; is being looked into today. Probably have a fix today or tomorrow depending on how things go.

**Comment by davidnuzik** [comment]
Blocked by #34088

**Comment by deniseschannon** [comment]
This should now be available to test

**Comment by davidnuzik** [comment]
Rancher version: v2.6-head `4def65610` 8/13/21
Rancher cluster type: single-node docker container
Docker version: 19.03

Downstream cluster type: RKE2 rancher-provisioned cluster (Digital Ocean node driver)
Downstream K8s version: v1.21.3-rc5+rke2r2

@deniseschannon Still not working, another new issue now

![image](https://user-images.githubusercontent.com/45179589/129400988-049ae432-5d5f-425a-a911-2383967900e3.png)

I created this issue to track: https://github.com/rancher/rancher/issues/34240

**Comment by davidnuzik** [comment]
Also UI behaves strangely when create any rke2 cluster with default config: https://github.com/rancher/dashboard/issues/3844

**Comment by thedadams** [comment]
@davidnuzik Does it work if you select an existing credential instead of providing one on cluster creation?

**Comment by davidnuzik** [comment]
@thedadams what does this mean exactly? I supplied the digitalocean cloud credential in the UI. Then, in cluster configuration section specified my s3 info

**Comment by davidnuzik** [comment]
Now I'm getting weird errors like this out of nowhere:
![image](https://user-images.githubusercontent.com/45179589/129416202-40dc3539-881e-4bd7-8c66-fbd54cc84cb3.png)

**Comment by thedadams** [comment]
@davidnuzik In the left side bar, you can click "Cloud Credentials" and create an s3-compatible cloud credential there. Then, when you create the cluster, you can choose the existing cloud credential that you created.

**Comment by davidnuzik** [comment]
@thedadams sorry for late follow up. Yes looks like if I manually create an s3-compatible cc first in the UI and use that to create the cluster I am able to do so then.

Side note: Interestingly, when I created the cluster with my s3-compatible cc with some optional defaults plugged in it didn't send those defaults in the web request. This is presumably a bug - will look into and file a separate issue for that as well.

**Comment by davidnuzik** [comment]
Reopened https://github.com/rancher/rancher/issues/34088.
If I don't specify the bucket manually on cluster creation form, even if specified in the s3-compatible cc, then cluster creation will fail.

**Comment by davidnuzik** [comment]
I was testing today and had everything all filled out (working around the existing bugs) but found https://github.com/rancher/rancher/issues/34272 this is now blocking this issue. Moving this issue back "To Test" for now while I await a fix.

**Comment by deniseschannon** [comment]
We have broken out this epic into 2 distinct issues. 

This one will be for testing local snapshot and restore. 

A new issue (https://github.com/rancher/rancher/issues/34417) was created to specifically track S3 support for snapshot and restore

**Comment by davidnuzik** [comment]
Moving back to blocked; smoke tested again and this still isn't working. Effectively blocked by https://github.com/rancher/rancher/issues/34365 local or s3 snapshot restoration will not work.

**Comment by davidnuzik** [comment]
v2.6-head 5f5f0ae57 pulled at 8/29/21 11:00 am Pacific

I am unblocked and have finished testing local snapshot/restore on RKE2 clusters via https://github.com/rancher/rancher/issues/34471 in that issue I tested a multi-node rke2 cluster - I tried several times to snapshot/restore - restore from automatic snapshots or on-demand and various options (each time you snapshot it takes as many as there are etcd nodes). I no longer encounter any issues with local rke2 snapshot/restore. All my tests were done with DigitalOcean since there is a bug with ec2 worker nodes currently - but it shouldn't make any difference the provider chosen.
